core/src/main/scala/kafka/metrics/clientmetrics/ClientMetricsCache.scala

```diff
@@ -131,8 +145,7 @@ class ClientMetricsCache(maxSize: Int) {
   }
 
   private def isExpired(element: CmClientInstanceState) = {
-    val currentTs = Calendar.getInstance.getTime
-    val delta = currentTs.getTime - element.getLastAccessTs.getTime
+    val delta = getCurrentTime - element.getLastAccessTs
     delta > Math.max(3 * element.getPushIntervalMs, DEFAULT_TTL_MS)
```

Comment:
I am interpreting this `3 * element.getPushIntervalMs` as:

>If we haven't seen a push from this client the last three times that we should have, boot it!

Am I understanding it correctly?

If a client has its information expires from this cache, does it get some kind of error, or is just an internal implementation on the broker?

Reply:
This is only for claiming the unused cache entries and  and it is internal implementation of the Broker's cache of the client instance objects. Even after the cache entry is deleted from the cache it would be recreated if there is any subsequent request from the client